kernel/signal.c

```diff
@@ -106,12 +109,42 @@ long myst_signal_sigprocmask(int how, const sigset_t* set, sigset_t* oldset)
     return ret;
 }
 
+void myst_signal_free_siginfos(myst_thread_t* thread)
+{
+    uint64_t unblocked = MYST_SIG_UNBLOCKED(thread->signal.mask);
+    uint64_t active_signals = thread->signal.pending & unblocked;
```

Comment:
@paulcallen There might be siginfos related to blocked signals. #420 is such a case.

Reply:
I am wondering if there are some race conditions that add after we finish freeing or something. I got a bunch of the cleanup where it was not so good before, but there seemed to be occasional times there was still one left behind.

Reply:
oh, blocked code came in after I implemented most of this! Now I understand!